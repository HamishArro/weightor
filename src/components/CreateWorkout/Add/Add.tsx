import React, {useState} from 'react';
import {View, Text, TextInput, Button} from 'react-native';
import {AddProps} from '../types';
import {styles} from '../../../utils/styleSheet';

function Add({handleAdd, handleBack}: AddProps) {
  const [name, setName] = useState<string>('');
  const [musclesUsed, setMusclesUsed] = useState<string>('');
  const [workoutEffort, setWorkoutEffort] = useState<string>('');
  const [duration, setDuration] = useState<string>('');

  const handleSubmit = () => {
    handleAdd({
      name,
      musclesUsed: musclesUsed.split(','),
      workoutEffort: parseInt(workoutEffort, 10),
      duration: parseInt(duration, 10),
    });
  };

  return (
    <View style={styles.stage} testID="add-container">
      <Text children={'Add exercise'} style={styles.title} />
      <Button testID="back-button" title="back" onPress={handleBack} />
      <TextInput
        testID="name-input"
        value={name}
        style={styles.textInput}
        onChangeText={setName}
        placeholder="please enter name here"
      />
      <TextInput
        testID="muscles-used-input"
        value={musclesUsed}
        style={styles.textInput}
        onChangeText={setMusclesUsed}
        placeholder="please enter muscles used"
      />
      <TextInput
        testID="workout-effort-input"
        value={workoutEffort}
        style={styles.textInput}
        onChangeText={setWorkoutEffort}
        placeholder="please enter workout effort"
        keyboardType="numeric"
      />
      <TextInput
        testID="duration-input"
        value={duration}
        style={styles.textInput}
        onChangeText={setDuration}
        placeholder="please enter duration"
        keyboardType="numeric"
      />
      <Button testID="add-button" title="add" onPress={handleSubmit} />
    </View>
  );
}

export default Add;
