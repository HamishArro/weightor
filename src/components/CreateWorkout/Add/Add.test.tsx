import * as React from 'react';
import {describe, it, expect, beforeEach, jest} from '@jest/globals';
import {fireEvent, render, screen, act} from '@testing-library/react-native';
import Add from './Add';

const props = {
  handleAdd: jest.fn(),
  handleBack: jest.fn(),
};

const cardioExercise = {
  name: 'cardio',
  musclesUsed: ['legs'],
  workoutEffort: 50,
  duration: 300,
};

describe('add tests', () => {
  beforeEach(() => {
    render(<Add {...props} />);
  });

  it('will call handle back when clicked', () => {
    act(() => fireEvent.press(screen.getByTestId('back-button')));

    expect(props.handleBack).toHaveBeenCalled();
  });

  it('can submit a form correctly', () => {
    act(() => {
      fireEvent.changeText(
        screen.getByTestId('name-input'),
        cardioExercise.name,
      );
      fireEvent.changeText(
        screen.getByTestId('muscles-used-input'),
        cardioExercise.musclesUsed.join(','),
      );
      fireEvent.changeText(
        screen.getByTestId('workout-effort-input'),
        JSON.stringify(cardioExercise.workoutEffort),
      );
      fireEvent.changeText(
        screen.getByTestId('duration-input'),
        JSON.stringify(cardioExercise.duration),
      );
    });

    act(() => fireEvent.press(screen.getByTestId('add-button')));

    expect(props.handleAdd).toHaveBeenCalledWith(cardioExercise);
  });

  it('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
