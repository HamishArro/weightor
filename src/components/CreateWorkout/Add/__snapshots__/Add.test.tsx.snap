// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add tests matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#dbdbdb",
      "width": "100%",
    }
  }
  testID="add-container"
>
  <Text
    style={
      {
        "fontSize": 24,
        "paddingBottom": 18,
      }
    }
  >
    Add exercise
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="back-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        back
      </Text>
    </View>
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="please enter name here"
    style={
      {
        "paddingBottom": 12,
      }
    }
    testID="name-input"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="please enter muscles used"
    style={
      {
        "paddingBottom": 12,
      }
    }
    testID="muscles-used-input"
    value=""
  />
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="please enter workout effort"
    style={
      {
        "paddingBottom": 12,
      }
    }
    testID="workout-effort-input"
    value=""
  />
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="please enter duration"
    style={
      {
        "paddingBottom": 12,
      }
    }
    testID="duration-input"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="add-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        add
      </Text>
    </View>
  </View>
</View>
`;
